hazelcast/src/main/java/com/hazelcast/query/impl/predicates/EvaluatePredicate.java

```diff
@@ -72,13 +72,23 @@ public boolean apply(Map.Entry mapEntry) {
     }
 
     @Override
-    public Set<QueryableEntry> filter(QueryContext queryContext) {
-        Index index = queryContext.matchIndex(indexName, QueryContext.IndexMatchHint.EXACT_NAME);
+    public Set<QueryableEntry> filter(QueryContext queryContext, int ownedPartitionCount) {
+        Index index = queryContext.matchIndex(indexName, QueryContext.IndexMatchHint.EXACT_NAME, ownedPartitionCount);
+        if (index == null) {
+            return null;
+        }
         return index.evaluate(predicate);
     }
 
     @Override
-    public boolean isIndexed(QueryContext queryContext) {
+    public boolean isIndexed(QueryContext queryContext, int ownedPartitionCount) {
+        if (predicate instanceof IndexAwarePredicate) {
```

Comment:
`EvaluatePredicate` already assume that we decided to use an index, so we should not check the nested predicate. Instead, we should use `EvaluatePredicate.indexName` with `EXACT_NAME` rule to check if the index is ready.